### ISSUE TITLE
new Seurat pre-processing code

### ISSUE THREAD

**Issue by daniel-spies** [LD-Lyndee/scQA#1]
Hi there,

thanks for the tool, excited to test it!

I ran your pre-processing code and noticed that it facilitates an older version of seurat, for newer versions count / logcounts methods don't work anymore. Either directly access the slots a@assays$RNA@counts a@assays$RNA@data or use the GetAssayData method.

same for setting it: SetAssayData ...

**Comment by LD-Lyndee** [comment]
Hello,

Thank you for your interest in my project. I appreciate your feedback and suggestions.

My test data was created as a SingleCellExperiment object, and count/logcounts slots can be accessed using count/logcounts methods. I noticed that this method will not work for a Seurat object, I will supplement the data preprocessing method for Seurat objects as you mentioned as soon as possible. The samples will also be provided in the Example folder.

**Comment by daniel-spies** [comment]
yeah I figured out then, it just wasn't immediately clear from the code. Another thing that is not fully documented, how exactly does the bin number affect the clustering? I would have guessed that more bins would allow for more clusters, but in my dataset it seems like higher numbers decrease the resolution. Also, do you have a recommendation regarding memory as with higher bins it seems to need a lot more of memory?

**Comment by LD-Lyndee** [comment]
Hi Daniel,

The number of bins is actually the number of pre-groups for gene clustering. Theoretically, the higher the number of bins, the higher the number of landmarks (number of clusters of genes) will be, but there should be no significant relationship with the number of clusters of cells. We normalized the data before clustering with all values in the interval [0, 1]. Based on this, we recommend using the default parameter which is the bin number set to 6. This parameter is relatively reasonable and we divide the [0, 1] interval into six subintervals, with [0, 0] being a single subinterval and the interval (0, 1] equivalently divided into the remaining five subintervals. 

Regarding memory consumption, I tested on several datasets, and increasing the number of bins could potentially reduce the peak memory consumption.
